apps/src/gamelab/animationListModule.js

```diff
@@ -687,28 +688,42 @@ function loadPendingFramesFromSource(key, props, callback) {
   };
 }
 
+
 /**
  * Given a key/serialized-props pair for an animation, work out where to get
  * the spritesheet.
  * @param {!AnimationKey} key
- * @param {!SerializedAnimationProps} props
+ * @param {object} animationList - Used to access props and check if src url
+ *        exists in this project and is deleted
+ * @param {string} channelId - Used to differentiate library animations from others
  * @returns {string}
  */
-export function animationSourceUrl(key, props) {
-  // TODO: (Brad) We want to get to where the client doesn't know much about
-  //       animation versions, by switching to Chris' new Files API.
-  //       in the meantime, be able to request versions only when we export
-  //       JSON for levelbuilders to use.
-
-  // 1. If the animation has a sourceUrl it's external (from the library
-  //    or some other outside source, not the animation API) - and we may need
-  //    to run it through the media proxy.
-  if (props.sourceUrl) {
+export function animationSourceUrl(key, animationList, channelId = "") {
```

Comment:
@islemaster - I don't think it makes sense for ChannelId to have a default option - if I use the empty string, then calls to animationSourceUrl without a channelId parameter are more permissive and those calls with a channelId parameter. If I use 'null', we have the opposite problem since it will check for 'v3/animations/null', which is always false. It is not possible to access channelId in all the places where animationSourceUrl is called, so making channelId required would be complicated (potentially more complicated than we'd like to pursue). Any ideas on how to handle this situation?

Reply:
It is possible to get the channelId everywhere `animationSourceUrl()` is called.  The only place it's not trivial is `withAbsoluteSourceUrls()`, called from `GameLab.prototype.getExportableAnimationList` where you'll also have access to `getStore()` - you'll just need to pass the channel ID through `withAbsoluteSourceUrls()`.

We should probably do some careful testing of the levelbuilder's case though - we want to make sure they can still correctly export animation URLs for use in levels.